ingestclient/utils/setup.py

```diff
@@ -0,0 +1,285 @@
+example, if the coordinate frame already exists, the properties specified in
+the JSON input file will be ignored.  If the Boss resource does not exist, it
+is created according to the specification in the input file.
+
+The boss-ingest config file that is referenced by this script's JSON input
+file is optionally updated with the name of the collection, experiment, and
+channel.  This property is "ingest_cfg" and is only required when using the
+`--writecfg` option.
+
+Below is a sample JSON input file for this script.  If a property is not marked
+with an "Optional" comment, it must be included in the JSON file.  In the 
+sample JSON, below, the values assigned to the optional properties are the
+default values that will be used if you do not provide these properties.  The
+exception is "ingest_cfg" as explained previously.
+
+{
+  "ingest_cfg": "ingest.json",      # boss-ingest config associated with this.
+  "collection": {
+    "name": "my_collection",
+    "description": "Example collection"
+  },
+  "coordinate_frame": {
+    "name": "my_coord",
+    "description": "Example coord frame",
+    "x_start": 0,
+    "x_stop": 1024,
+    "y_start": 0,
+    "y_stop": 1024,
+    "z_start": 0,
+    "z_stop": 32,
+    "voxel_size_x": 3,
+    "voxel_size_y": 3,
+    "voxel_size_z": 30
+  },
+  "experiment": {
+    "name": "my_exp",
+    "description": "Example experiment",
+    "num_time_samples": 1,                  # Optional.
+    "num_hierarchy_levels": 1,              # Optional.
+    "hierarchy_method": "anisotropic",      # Optional.
+    "time_step": 0,                         # Optional.
+    "time_step_unit": "seconds"             # Optional.
+  },
+  "channel": {
+    "name": "my_chan",
+    "description": "Example channel",
+    "type": "image",
+    "datatype": "uint8",
+    "sources": []                           # Optional names of source channels.
+  }
+}
+"""
+
+import argparse
+from intern.remote.boss import BossRemote
+from intern.resource.boss.resource import *
+import json
+import os
+
+
+# Name of group to assign read-only privileges to.
+GOV_TEAM = 'gov_team'
+
+def create_collection(config, rmt):
+    """
+    Ensure collection exists.
+
+    Args:
+        config (dict):
+        rmt (BossRemote):
+
+    Returns:
+        (CollectionResource)
+    """
+    collection = CollectionResource(
+        config['collection']['name'], config['collection']['description'])
+    try:
+        collection = rmt.create_project(collection)
+    except Exception as e:
+        collection = rmt.get_project(collection)
+    return collection
+
+
+def create_experiment(config, rmt, collection, coord):
+    """
+    Ensure channel exists.
+
+    Args:
+        config (dict):
+        rmt (BossRemote):
+        collection (CollectionResource):
+        coord (CoordinateFrameResource):
+
+    Returns:
+        (ExperimentResource)
+    """
+    if 'num_time_samples' not in config['experiment']:
```

Comment:
Instead of having all of these statements it would be easier to create a dictionary of default values and update with the actual values.

```py
exp = {'num_time_samples': 1, } # add all default values
exp.extend(config['experiment'])
# now `exp` contains the experiment configuration with default values overridden
```

Reply:
Thanks, that's much more elegant than a bunch of if/else statements!